app/jobs/scan/library/detect_filesystem_changes_job.rb

```diff
@@ -39,26 +39,25 @@ def folders_with_changes(library)
     disk_files = disk_files.to_a unless disk_files.is_a?(Array)
 
     # Find files on disk that aren't in database (new files)
-    # Process in batches to reduce peak memory usage when checking against set
+    # Process in batches to reduce peak memory usage
     disk_files.each_slice(BATCH_SIZE) do |batch|
       batch.each do |path|
-        # Skip thingiverse false-positives
         next if thingiverse_patterns.any? { |pattern| path =~ pattern }
 
         unless known_paths.include?(path)
-          # File exists on disk but not in database - it's new
           changed_folders.add(File.dirname(path))
         end
       end
     end
 
     # Find files in database that aren't on disk (deleted files)
-    # Process known_paths in batches to reduce peak memory usage
+    # Process in batches to reduce peak memory usage
     disk_files_set = disk_files.to_set
     known_paths.each_slice(BATCH_SIZE) do |batch|
       batch.each do |path|
+        next if thingiverse_patterns.any? { |pattern| path =~ pattern }
```

Comment:
This is unnecessary; while the thingiverse false positive filtering is useful in the "new files" scan, in the "removed files" part it will do absolutely nothing, as those files wouldn't have been added to Manyfold in the first place.